sed/calibrator/delay.py

```diff
@@ -25,22 +28,102 @@ class DelayCalibrator:
     def __init__(
         self,
         config: dict = None,
-    ):
+    ) -> None:
         """Initialization of the DelayCalibrator class passes the config.
 
         Args:
             config (dict, optional): Config dictionary. Defaults to None.
         """
-        if config is None:
-            config = {}
-
-        self._config = config
+        self._config: dict = config or {}
+        self.loader: str = self._config["core"]["loader"]
 
-        self.adc_column = self._config["dataframe"]["adc_column"]
-        self.delay_column = self._config["dataframe"]["delay_column"]
-        self.calibration: Dict[Any, Any] = {}
+        self.adc_column: str = self._config["dataframe"].get("adc_column", None)
+        self.delay_stage_column: str = self._config["dataframe"].get("delay_stage_column", None)
+        if self.delay_stage_column is None and self.adc_column is None:
+            raise ValueError("No delay stage column specified.")
+        self.delay_column: str = self._config["dataframe"]["delay_column"]
+        self.calibration: Dict[str, Any] = self._config["delay"].get("calibration", {})
+        self.fluctuations: Dict[str, Any] = self._config["delay"].get("fluctuations", {})
 
     def append_delay_axis(
+        self,
+        df: Union[pd.DataFrame, dask.dataframe.DataFrame],
+        *args,
+        **kwargs,
+    ) -> Tuple[Union[pd.DataFrame, dask.dataframe.DataFrame], dict]:
+        """TODO: docstring"""
+        if self.loader not in ["mpes", "flash"]:
+            raise NotImplementedError(
+                f"Delay calibration is implemented for 'mpes' and 'flash', not {self.loader}.",
+            )
+        method = getattr(self, f"append_delay_axis_{self.loader}")
+        return method(df, *args, **kwargs)
```

Comment:
Is it necessary to do this overloading of the function? For flash, you do not strictly do a calibration (i.e. changing of coordinate system) of the delay axis, but essentially a shift, and this would be a function also interesting to have for mpes (if the calibration was off or so). So I would split the two into two seperate functions, calibrate_delay_axis, and shift_delay_axis.